lib/Nexus.js

```diff
@@ -268,6 +269,42 @@ module.exports = function xGraph(__options = {}) {
 				return pid;
 			}
 
+			/**
+			 * Scan the array of Pids and send a 'GetServices'
+			 * command to each in turn to accumulate the 
+			 * services used by a partifcular module
+			 * @param {array} services		array of Pids to send GetServices to
+			 */
+			async function getServices(services) {
+				let svc = {};
+				let com = {};
+				
+				return new Promise(async (resolve, _reject) => {
+					for (let is = 0; is < services.length; is++) {
+						let pid = services[is];
+						await service(pid);
+					}
+					resolve(svc);
+				});
+
+				async function service(pid) {
+					return new Promise((res, _reg) => {
+						com.Cmd = 'GetServices';
+						com.Passport = {};
+						com.Passport.To = pid;
+						com.Passport.Pid = genPid();
+						sendMessage(com, function(_err, r) {
```

Comment:
using messaging as the core for acquiring services at runtime, defeats the purpose of moving them to nexus level. Doing so, still allows for services from a module to change at runtime. This works as an intermediary solution (in a branch, as it is), however it is not develop ready, as it is not the changes we agreed on.